tests/contenttypes_tests/tests.py

```diff
@@ -428,3 +429,43 @@ def test_multidb(self):
         with self.assertNumQueries(0, using='default'), \
                 self.assertNumQueries(1, using='other'):
             ContentType.objects.get_for_model(Author)
+
+
+class TestContentTypeRenaming(TransactionTestCase):
+    available_apps = ['django.contrib.contenttypes', 'contenttypes_tests']
+
+    @override_settings(MIGRATION_MODULES={'contenttypes_tests': 'contenttypes_tests.rename_migrations'})
+    def test_rename_model_renames_content_type_model(self):
+        """
+        A RenameModel operation should result in a ContentType.model rename if
+        it exists.
+        """
+        def assert_renames_inserted(plan, **kwargs):
+            for migration, _backward in plan:
+                operations = iter(migration.operations)
+                for operation in operations:
+                    if isinstance(operation, migrations.RenameModel):
+                        next_operation = next(operations)
+                        self.assertIsInstance(next_operation, contenttypes_management.RenameContentType)
+                        self.assertEqual(next_operation.app_label, migration.app_label)
+                        self.assertEqual(next_operation.old_name, operation.old_name)
+                        self.assertEqual(next_operation.new_name, operation.new_name)
+        app_config = apps.get_app_config('contenttypes_tests')
+        models.signals.post_migrate.connect(assert_renames_inserted, sender=app_config)
+        stdout = six.StringIO()
+        management.call_command(
+            'migrate', 'contenttypes_tests', database='default', interactive=False, stdout=stdout
+        )
+        # Models with an existing contenttype should be renamed.
+        self.assertFalse(ContentType.objects.filter(app_label='contenttypes_tests', model='foo').exists())
+        self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedfoo').exists())
+        # Models with no prior contenttypes should't be renamed.
+        self.assertFalse(ContentType.objects.filter(app_label='contenttypes_tests', model='bar').exists())
+        self.assertFalse(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedbar').exists())
+        # Create a stale content type to make sure conflicts are correctly handled.
+        ContentType.objects.create(app_label='contenttypes_tests', model='foo')
```

Comment:
@timgraham, the creation of this stale content type should trigger the issue you described.